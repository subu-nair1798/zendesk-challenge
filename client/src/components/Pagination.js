import React from 'react';

const Pagination = ({ ticketsPerPage, totalTickets, paginate }) => {
    const pageNumbers = [];

    for(let i = 1; i <= Math.ceil(totalTickets / ticketsPerPage); i++) {
        pageNumbers.push(i);
    }

    return (
        <nav>
            <ul className="pagination">
                {pageNumbers.map(number => (
                    <li key={number} className="page-item">
                        <div style={{ cursor: "pointer" }} onClick={() => paginate(number)} className="page-link">
                            {number}
                        </div>
                    </li>
                ))}
            </ul>
        </nav>
    );
}

export default Pagination;