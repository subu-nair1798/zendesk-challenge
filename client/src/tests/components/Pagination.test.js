import React from 'react';
import { shallow } from 'enzyme';
import Pagination from '../../components/Pagination';
import { sampleResponse } from '../sampleResponse';
import { tickets } from '../../../../tickets.json';

describe('<Pagination />', () => {
    it('render Pagination component', () => {
        const paginate = (pageNumber) => this.setState({ currentPage: pageNumber });
        const wrapper = shallow(<Pagination ticketsPerPage={25} totalTickets={sampleResponse.tickets.length} paginate={paginate} />);
        expect(wrapper).toMatchSnapshot();
    });

    it('should create a single page button for a sample response of 10 tickets', () => {
        const paginate = (pageNumber) => this.setState({ currentPage: pageNumber });
        const wrapper = shallow(<Pagination ticketsPerPage={25} totalTickets={sampleResponse.tickets.length} paginate={paginate} />);
        expect(wrapper.find('.pagination').children()).toHaveLength(1);
    });

    it('should create 4 page buttons for a sample response of 100 tickets', () => {
        const paginate = (pageNumber) => this.setState({ currentPage: pageNumber });
        const wrapper = shallow(<Pagination ticketsPerPage={25} totalTickets={tickets.length} paginate={paginate} />);
        expect(wrapper.find('.pagination').children()).toHaveLength(4);
    });
});