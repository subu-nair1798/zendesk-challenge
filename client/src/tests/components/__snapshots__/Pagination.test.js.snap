// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination /> render Pagination component 1`] = `
<nav>
  <ul
    className="pagination"
  >
    <li
      className="page-item"
      key="1"
    >
      <div
        className="page-link"
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer",
          }
        }
      >
        1
      </div>
    </li>
  </ul>
</nav>
`;
